content/kb/troubleshooting-rwx-volume-migration-stuck.md

```diff
@@ -0,0 +1,149 @@
+
+Example volume state:
+
+```yaml
+Volume: pvc-abcdefg
+  spec.nodeID: s1
+  status:
+    robustness: degraded
+    state: attached
+
+  Engine:
+    name: pvc-abcdefg-e-1
+    spec.nodeID: s1
+    status:
+      currentState: running
+      currentReplicaAddressMap:
+        - pvc-abcdefg-r-5d917bb3: 10.52.8.201:11840
+        - pvc-abcdefg-r-fe42a309: 10.52.2.101:11812
+
+    name: pvc-abcdefg-e-2
+    spec.nodeID: t2
+    status:
+      currentState: running
+      currentReplicaAddressMap:
+        - pvc-abcdefg-r-3f99a289: 10.52.2.101:11823
+        - pvc-abcdefg-r-aa3ef1d9: 10.52.8.201:11850
+```
+
+onghorn Volume CR indicating migration is in progress:
+
+```bash
+$ kubectl describe lhv pvc-abcdefg
+```
+
+```yaml
+apiVersion: longhorn.io/v1beta2
+kind: Volume
+metadata:
+  labels:
+    longhornvolume: pvc-abcdefg
+  name: pvc-abcdefg
+  namespace: longhorn-system
+  ...
+spec:
+  accessMode: rwx
+  backingImage: default-image-klmt7
+  dataEngine: v1
+  image: longhornio/longhorn-engine:v1.7.3
+  migratable: true
+  migrationNodeID: t2
+  nodeID: s1
+  numberOfReplicas: 3
+  ...
+status:
+  currentMigrationNodeID: t2
+  currentNodeID: s1
+  ownerID: s1
+  robustness: degraded
+  state: attached
+  ...
+```
+
+VolumeAttachment CR confirms that the migration ticket has been satisfied:
+
+```bash
+$ kubectl -n longhorn-system describe lhva pvc-abcdefg
+```
+
+```yaml
+Name:         pvc-abcdefg
+Namespace:    longhorn-system
+Kind:         VolumeAttachment
+Spec:
+  Attachment Tickets:
+    csi-473853237b61d7ea80ea8f3b9306d82c55ccf36744ee88212ee95c4c2f299edb:
+      Type:                csi-attacher
+      ...
+    csi-bf34f58ee9ac935d1120e60253c7a4f9c1e73afc411677278848d0f1bcaace96:
+      Type:                csi-attacher
+      ...
+  Volume:                  pvc-abcdefg
+Status:
+  Attachment Ticket Statuses:
+    csi-473853237b61d7ea80ea8f3b9306d82c55ccf36744ee88212ee95c4c2f299edb:
+      Conditions:
+        Last Transition Time:  2025-06-19T07:24:07Z
+        Message:               The migrating attachment ticket is satisfied
+        Status:                True
+        Type:                  Satisfied
+      Generation:              0
+      Id:                      csi-473853237b61d7ea80ea8f3b9306d82c55ccf36744ee88212ee95c4c2f299edb
+      Satisfied:               true
+    csi-bf34f58ee9ac935d1120e60253c7a4f9c1e73afc411677278848d0f1bcaace96:
+      Conditions:
+        Last Transition Time:  2025-06-19T06:01:18Z
+        Status:                True
+        Type:                  Satisfied
+      Generation:              0
+      Id:                      csi-bf34f58ee9ac935d1120e60253c7a4f9c1e73afc411677278848d0f1bcaace96
+      Satisfied:               true
+```
+
+## Reason
+
+There are cases where Longhorn fails to finalise RWX volume migration even after successful attachment on the destination node. This incomplete migration blocks Harvester VM live migration, leaving it stuck in the pre-drain stage.
```

Comment:
```suggestion
There are cases where Longhorn fails to finalise RWX volume migration even after successful attachment on the destination node. This incomplete migration blocks Harvester VM live migration, leaving it stuck in the pre-drain stage.
```

It is due to the potential issue in kubelet. Can you add the information of this and the related messages we have seen in the support bundle?